### PR TITLE
feat(network): apply host firewall (inet host) during host provisioning

### LINKED ISSUE #778: Story 0.1 — Wire `network firewall create` into `kube cluster install` (node-agnostic)
**Epic:** #777 — Host network firewall (node-agnostic)

Design: [v4 design](https://github.com/automa-saga/traffic-shaper/blob/main/docs/bn-qos-multiclass-priority-design-v4-nft-priority.md) §8.4.1.

Wire `network firewall create` into the node-agnostic host-provisioning path so every node type (block, consensus, mirror, relay) gets the `inet host` table laid down once per machine, independent of any workload install.

**Acceptance criteria:**
- Invoke `network firewall create` (the generic command, #757) from `kube cluster install` → `systemSetupWorkflow()` (`internal/workflows/setup.go`), after the existing nftables systemd-service step.
- Runs for all node types (not gated on `block`); idempotent (create-if-missing).
- Ordering: the `inet host` base + SSH/mgmt allow rules are applied before any default-drop takes effect, so a host is never briefly locked out during provisioning.
- The BN workload plane (`inet weaver`) is NOT touched here — that stays with `block node install` (TS_2 #762).

### LINKED ISSUE #779: Story 0.2 — Host mgmt-CIDR config: `--mgmt-cidrs` flag + profile default
**Epic:** #777 — Host network firewall (node-agnostic)

Design: [v4 design](https://github.com/automa-saga/traffic-shaper/blob/main/docs/bn-qos-multiclass-priority-design-v4-nft-priority.md) §8.4.1.

Provide the management-CIDR config surface for the host firewall. The host's own SSH/mgmt access is solo-provisioner/host-level config (it protects the bare-metal host the provisioner runs on) — it is NOT installed-component config, so it does not belong in a component's values.

**Acceptance criteria:**
- Add `--mgmt-cidrs` (and `--ssh-port`, ICMP policy) to `kube cluster install`, forwarded to `network firewall create`.
- Profile-driven default: a host-level `HostConfig.ManagementCIDRs` (`pkg/models/config.go`); profiles are global per-machine, so the default applies to whatever node type the host runs.
- Day-2 changes are operator-driven via the generic `network firewall add/remove/set --mgmt-cidr` verbs (#757) — no re-install required.
- ICMP policy (`--icmp-mgmt`/`--icmp-public`) depends on the same mgmt-CIDR set (planning note 9).
- Document the ownership: mgmt CIDRs are a host/cluster-install input, not a `block node install` input.

### PR DESCRIPTION
**Pull request by brunodam**
## Summary

Wires `network firewall create` (#757) into the **block-node workflow** so every block node gets the node-level `inet host` nftables table — SSH/management allowlist, ICMP policy, and in-cluster host-service ports — laid down before its Helm chart deploys. It is create-if-missing (re-running install is a no-op).

This PR **closes #778 and #779**. #779 (the management-CIDR config surface) is folded in because the `inet host` input chain is default-drop: applying it with an empty SSH allowlist would permanently lock a host out of new SSH connections. Since this branch targets `main` directly (no epic rollup branch), #778 must be independently safe there, which requires #779's allowlist plumbing.

### Design correction from the original approach

The first version of this PR wired the firewall step into the generic `systemSetupWorkflow()` (shared by `kube cluster install` and bare-metal `block node install`), following epic #777's original framing that this is "node-agnostic" host provisioning. Feedback surfaced that this is **too invasive for `kube cluster install`**, which is used as a general-purpose cluster-provisioning tool for varying deployments — forcing node-specific firewall rules onto every generic cluster install doesn't fit that use case. Having it live in the block-node (and, in future, consensus-node) workflow instead makes more sense, since that's the actual node type that needs the protection today.

This version moves the wiring accordingly:

- `internal/workflows/setup.go` — `NetworkFirewallCreate()` **removed** from `systemSetupWorkflow()` (still installs/enables nftables itself, since other things may depend on the package being present — just no longer applies the `inet host` table there).
- `cmd/cli/commands/kube/cluster/install.go` — host-firewall flags/resolution **removed**. `kube cluster install` no longer touches the firewall at all.
- `internal/bll/blocknode/install_handler.go` — `NetworkFirewallCreate()` **prepended** to both branches of `InstallHandler.BuildWorkflow` (bare-metal bootstrap and deploy-onto-existing-cluster). Both branches are guaranteed to already have nftables installed by the time this step runs (either from a prior `kube cluster install`, or from the cluster bootstrap this same `block node install` just ran).
- `cmd/cli/commands/block/node/install.go` — now **always** resolves the host firewall config (previously gated on "bootstrapping bare metal only" via a `state.ReadClusterCreatedFromDisk()` check, since the firewall was expected to already exist for the existing-cluster case; that gate and its now-dead reader are removed).

The corresponding traffic-shaper design doc and epic #777 are being updated to match (tracked separately).

### What changed

- **Host firewall step** (`internal/workflows/steps/step_network_firewall.go`): `NetworkFirewallCreate` step. Reads the resolved host config, calls `firewall.NewManager().Create(...)`. **Skips with a loud warning when no management CIDRs are configured** rather than rendering a lock-out ruleset, and **skips (info-level) when explicitly disabled** via `--firewall-enabled=false`. Rollback deletes the table only if this step created it.
- **Config surface** (`pkg/models/config.go`, `pkg/config/global.go`): new `host` config block — `managementCidrs`, `sshPort`, `podCidr`, `inClusterPorts`, `disabled` — with validation, plus `OverrideHostConfig` (full-replace semantics, since the resolver already computes the complete effective state before calling it).
- **CLI flags + prompts + shared resolver** (`cmd/cli/commands/common/host_firewall.go`): `--firewall-enabled`, `--mgmt-cidrs`, `--ssh-port`, `--pod-cidr`, `--in-cluster-ports`, registered via `RegisterHostFirewallFlags` and resolved via `ResolveHostFirewallConfig`. Precedence per value: CLI flag > interactive prompt > config file > built-in default. Interactive installs prompt (pre-filled, confirm with Enter) for any value not passed on the CLI; `--non-interactive` / non-TTY skips prompting. `--firewall-enabled=false` opts out entirely and skips resolving the other fields.
- **`block node install` wired**: always resolves and applies, regardless of bare-metal vs. existing-cluster.
- **Single source of truth for the pod subnet**: promoted `10.4.0.0/14` to `models.DefaultClusterPodCIDR`, now consumed by both the kubeadm `podSubnet` template (`KubeadmInitData.PodSubnet`) and the firewall `--pod-cidr` default — so the in-cluster host-service ports rule opens exactly the range kubeadm assigns pods (the per-node CIDR can't be auto-detected before the cluster exists).

### Scope boundaries (epic #777)

- **Not** touched here: the BN `inet weaver` workload plane (#762), teardown on `kube cluster uninstall` (#791, itself likely needs the same block-node-vs-cluster ownership correction). Reboot persistence of `network-host.nft` + the shared oneshot unit is already handled by #757's `Create` path; #780 refines it.
- **Follow-up story** (#805) wires the same firewall step into `block node reconfigure` / `upgrade` (opt-in, not automatic) so hosts already running a pre-#802-era binary can retrofit the firewall without a fresh install.

## Test plan

- [x] `task lint:check` — 0 issues
- [x] `go build` / `go vet` (GOOS=linux) — clean across all touched packages
- [x] Unit (macOS): `go test ./pkg/models/... ./internal/ui/prompt/... ./pkg/config/... ./internal/state/...`
  - `HostConfig.Validate` cases; prompt validators (`mgmt-cidrs` / `ssh-port` / `pod-cidr` / `in-cluster-ports`) + `ParsePortList`; `OverrideHostConfig` full-replace semantics
- [ ] Unit (VM, Linux-only pkg): `task vm:test:unit` covering `internal/workflows/steps` — new step tests (skip-on-empty, skip-on-disabled, create-on-set + rollback-deletes, rollback-skips-when-preexisting, explicit-empty overrides defaults)
- [ ] Integration (VM): `task vm:test:integration` — cluster-install / block-node-install path
- [ ] Manual UAT on a host with real nft (see step-by-step below)

### Manual UAT (host with real nftables)

Run on a provisioning host (or the UTM VM). Keep a second SSH session open as a safety net before applying default-drop.

**1. Happy path — firewall applied with a management allowlist, on top of an existing cluster**

```bash
sudo solo-provisioner kube cluster install --profile=local --node-type=block --non-interactive
sudo solo-provisioner block node install --profile=local \
  --non-interactive --mgmt-cidrs=10.0.0.0/8 --pod-cidr=10.4.0.0/14
```

Then inspect the live table:

```bash
sudo nft list table inet host
```

Expected (abridged) — SSH allow scoped to the allowlist, the in-cluster-ports rule for the pod CIDR, and `policy drop`:

```
table inet host {
        set mgmt_addrs { type ipv4_addr; flags interval; elements = { 10.0.0.0/8 } }
        set in_cluster_ports { type inet_service; elements = { 4244, 6443, 7472, 10250 } }
        chain input {
                type filter hook input priority filter; policy drop;
                ...
                ip saddr @mgmt_addrs tcp dport 22 accept
                ip saddr 10.4.0.0/14 tcp dport @in_cluster_ports accept
        }
}
```

Confirm the on-disk artifact and the boot unit exist, and that SSH from a management source still connects:

```bash
cat /etc/solo-provisioner/network-host.nft        # same ruleset as above
systemctl is-enabled solo-provisioner-network-nft.service   # -> enabled
```

**2. `kube cluster install` alone never touches the firewall**

```bash
sudo solo-provisioner kube cluster install --profile=local --node-type=block --non-interactive
sudo nft list table inet host
```

Expected: `Error: No such file or directory` — the table is never created by `kube cluster install` on its own, confirming the ownership move.

**3. Bare-metal bootstrap via `block node install`**

On a machine with no cluster yet:

```bash
sudo solo-provisioner block node install --profile=local \
  --non-interactive --mgmt-cidrs=10.0.0.0/8
sudo nft list table inet host     # same ruleset as scenario 1
```

Expected: the bootstrap path (cluster + block node in one command) applies the firewall.

**4. Idempotency — re-running install is a no-op**

```bash
sudo solo-provisioner block node install --profile=local \
  --non-interactive --mgmt-cidrs=10.0.0.0/8
```

Expected: a warning that the table already exists and flags were not re-applied (create-if-missing); the step reports success without changing the table.

**5. Skip path — no allowlist, no lockout**

```bash
sudo solo-provisioner block node install --profile=local --non-interactive
```

Expected: the step is **skipped** with a warning, and no table is created:

```
WARN host firewall not applied: no management CIDRs configured ...
$ sudo nft list table inet host
Error: No such file or directory   # table was never created -> host not locked out
```

**6. Explicit opt-out**

```bash
sudo solo-provisioner block node install --profile=local \
  --non-interactive --mgmt-cidrs=10.0.0.0/8 --firewall-enabled=false
```

Expected: the step is skipped (info-level, not a warning — this is an intentional choice, not a safety fallback), and no table is created even though `--mgmt-cidrs` was supplied.

**7. Interactive prompt — values pre-filled, confirm with Enter**

Run without `--non-interactive` and without the firewall flags on a TTY:

```bash
sudo solo-provisioner block node install --profile=local
```

Expected: four prompts appear (Management CIDRs, SSH port `22`, Pod CIDR `10.4.0.0/14`, In-cluster ports `6443,4244,7472,10250`), each pre-filled and acceptable with Enter; a "Host Firewall" summary of the chosen values is printed before the workflow runs.

**8. kubeadm podSubnet unchanged**

```bash
grep podSubnet /opt/solo-provisioner/.../etc/weaver/kubeadm-init.yaml   # -> podSubnet: 10.4.0.0/14
```

Expected: identical to the previous literal — the constant change is behavior-neutral.

## Risks / rollback

- **SSH lockout** — mitigated: empty `--mgmt-cidrs` → step skips (no drop policy applied); the SSH allow rule and the drop policy commit in one atomic `nft -f` transaction (no window).
- **Default-drop active before kubeadm init (bare-metal bootstrap path)** — the step runs after `InstallClusterWorkflow`'s cluster bootstrap but before `SetupBlockNode`. Loopback and established/related are accepted, so single-node bootstrap (predominantly loopback) is unaffected. **Multi-node** control-plane/worker traffic to `6443` originates from node IPs, so those node subnets must be included in `--mgmt-cidrs` (the in-cluster-ports rule opens the pod CIDR only). Reviewer attention welcome here.
- **kubeadm template change** — `podSubnet` now renders from `models.DefaultClusterPodCIDR` (value identical to the prior literal `10.4.0.0/14`); behavior-neutral. Startup migrations do not re-render kubeadm config, so existing clusters are unaffected.
- Rollback: revert the `install_handler.go` step insertion; the step's own rollback deletes a table it created. Additive otherwise — no existing path invoked it before.

Closes #778
Closes #779

### CODE REVIEW

File: cmd/cli/commands/kube/cluster/install.go [hunk 1]
```diff
@@ -80,11 +93,147 @@ var installCmd = &cobra.Command{
...
+		SSHPort:         sshPort,
+		PodCIDR:         strings.TrimSpace(podStr),
+		InClusterPorts:  ports,
+	}
+	if err := hostCfg.Validate(); err != nil {
+		return errorx.IllegalArgument.Wrap(err, "invalid host firewall configuration")
+	}
+	config.OverrideHostConfig(hostCfg)
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (brunodam) [reply cmd/cli/commands/kube/cluster/install.go]: Fixed in c34e990: `OverrideHostConfig` now replaces the host config wholesale instead of skipping empty/nil fields — the only caller already resolves the full flag > prompt > config file > default precedence before calling it, so an explicit empty value is always the caller's real intent, not a partial patch.

File: cmd/cli/commands/kube/cluster/install.go [hunk 2]
```diff
@@ -80,11 +93,147 @@ var installCmd = &cobra.Command{
...
+func effectiveIntCSV(cmd *cobra.Command, name string, flagVal, cfgVal, def []int) string {
+	switch {
+	case cmd.Flags().Changed(name):
+		return joinInts(flagVal)
+	case len(cfgVal) > 0:
+		return joinInts(cfgVal)
+	default:
+		return joinInts(def)
+	}
+}
```
[the remark this answers is not in the document]
Comment (brunodam) [reply cmd/cli/commands/kube/cluster/install.go]: Fixed in c34e990 along with the sibling comment — `OverrideHostConfig` no longer special-cases empty/nil, so `host.inClusterPorts: []` now correctly clears the default port set.

File: internal/workflows/steps/step_network_firewall.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			// port set); override each field only when the resolved host config
+			// carries a value, so a partial config never wipes a good default.
+			t := firewall.NewTable()
+			t.MgmtCIDRs = hostCfg.ManagementCIDRs
+			if hostCfg.SSHPort != 0 {
+				t.SSHPort = hostCfg.SSHPort
+			}
+			if len(hostCfg.InClusterPorts) > 0 {
+				t.InClusterPorts = hostCfg.InClusterPorts
+			}
```
[the remark this answers is not in the document]
Comment (brunodam) [reply internal/workflows/steps/step_network_firewall.go]: Fixed in c34e990: dropped the `len(hostCfg.InClusterPorts) > 0` guard so the field is applied unconditionally (matching how PodCIDR/MgmtCIDRs are already handled here) — an explicitly empty list now renders with no in-cluster ports. Added a regression test (`TestNetworkFirewallCreate_ExplicitEmptyPortsAndPodCIDROverrideDefaults`) asserting the rendered nft file has no default ports/rule when both are explicitly cleared.

**Review by alex-au — DISMISSED** [review]
LGTM!

Just a claude nitpick for consideration:

> - pkg/models/config.go:994-1016 (HostConfig.Validate) and internal/ui/prompt/cluster.go:376-391 (validateMgmtCIDRs) — new CIDR validation is looser than the firewall layer it feeds, causing a late, expensive failure instead of an immediate one. Both call only sanity.ValidateCIDR, which accepts IPv6 CIDRs (plain net.ParseCIDR). But internal/network/firewall/table.go:59-68 (Table.Validate, called inside Manager.Create, unchanged by this PR) additionally rejects any CIDR where ip.To4() == nil, since the inet host table only supports ipv4_addr sets. The same gap applies to PodCIDR (validatePodCIDR at internal/ui/prompt/cluster.go:395-404, and HostConfig.Validate's PodCIDR branch).
> - Failure scenario: an operator runs block node install --mgmt-cidrs=2001:db8::/32 ... on a fresh (bare-metal) host. The prompt validator and ResolveHostFirewallConfig's hostCfg.Validate() both accept the value and it's persisted via OverrideHostConfig. The command proceeds through InstallHandler.BuildWorkflow's bare-metal branch, which runs the entire InstallClusterWorkflow (kubelet, cri-o, kubeadm init, cilium, MetalLB — everything) before reaching steps.NetworkFirewallCreate() (internal/bll/blocknode/install_handler.go:318-326). Only then does firewall.Manager.Create call Table.Validate() and reject the IPv6 CIDR — failing the workflow after a full, non-trivial cluster bootstrap has already completed, for input that should have been rejected at the very first prompt.
> - Fix: add the same ip.To4() == nil check to HostConfig.Validate and the two prompt validators (or factor it into a shared sanity helper both layers call), so this is caught at input time.

**Review by alex-au — APPROVED** [review]
Nice!
